### PR TITLE
Bug 1828611: Add events for Noobaa for activity card

### PR DESCRIPTION
**Pull request by a2batic**
Add events for Noobaa Backing store, Noobaa Bucket class, OBC and OB for Object dashboard activity card

Screenshot from 2020-04-28 18-26-54

Fixes: https://issues.redhat.com/browse/RHSTOR-1022
Signed-off-by: Kanika <kmurarka@redhat.com>

cc: @afreen23

### CODE REVIEW

File: frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx [hunk 1]
```diff
@@ -19,17 +19,24 @@ import {
...
 import { DATA_RESILIENCE_QUERIES } from '../../queries';
 import './activity-card.scss';
+import { NooBaaBackingStoreModel, NooBaaBucketClassModel } from '../../models';
 
 const eventsResource: FirehoseResource = { isList: true, kind: EventModel.kind, prop: 'events' };
 
 const isNoobaaEventObject = (event: EventKind): boolean => {
   const eventName: string = _.get(event, 'involvedObject.name');
-  return _.startsWith(eventName, 'noobaa');
+  const component: string = _.get(event, 'source.component');
```
Comment (rawagner) [inline frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx:28]: optional chaining is preferred, even for the line below

**Comment by afreen23** [comment]
/retitle Add events for Noobaa for activity card

**Comment by afreen23** [comment]
/lgtm cancel

Just to fix this, cancelling.
Screenshot from 2020-04-24 19-46-06

File: frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx [hunk 2]
```diff
@@ -19,17 +19,24 @@ import {
 import { getResiliencyProgress } from '@console/ceph-storage-plugin/src/utils';
 import { DATA_RESILIENCE_QUERIES } from '../../queries';
 import './activity-card.scss';
+import { NooBaaBackingStoreModel, NooBaaBucketClassModel } from '../../models';
```
Comment (afreen23) [inline frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx:22]: Nit:
Can you fix the order of imports ?

File: frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx [hunk 3]
```diff
@@ -19,17 +19,24 @@ import {
...
+  return _.startsWith(eventName, 'noobaa') || _.startsWith(component, 'noobaa');
 };
 
 const noobaaEventsFilter = (event: EventKind): boolean => {
   const eventKind: string = _.get(event, 'involvedObject.kind');
-  if (eventKind === PodModel.kind || eventKind === StatefulSetModel.kind) {
+  if (
+    eventKind === PodModel.kind ||
+    eventKind === StatefulSetModel.kind ||
+    eventKind === NooBaaBackingStoreModel.kind ||
```
Comment (afreen23) [inline frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx:37]: Backing store is a Noobaa entity, `isNoobaaEventObject(event)` can be avoided for this.

File: frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx [hunk 4]
```diff
@@ -19,17 +19,34 @@ import {
...
 
 const isNoobaaEventObject = (event: EventKind): boolean => {
-  const eventName: string = _.get(event, 'involvedObject.name');
-  return _.startsWith(eventName, 'noobaa');
+  const eventName: string = event?.involvedObject?.name;
+  const component: string = event?.source?.component;
+  const eventKind: string = event?.involvedObject?.kind;
+  return (
+    _.startsWith(eventName, 'noobaa') ||
+    _.startsWith(component, 'noobaa') ||
```
Comment (afreen23) [inline frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx:36]: Why are we using `component` and `eventName` both here ?

File: frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx [hunk 5]
```diff
@@ -19,17 +19,34 @@ import {
 import { getResiliencyProgress } from '@console/ceph-storage-plugin/src/utils';
 import { DATA_RESILIENCE_QUERIES } from '../../queries';
 import './activity-card.scss';
+import {
+  NooBaaBackingStoreModel,
+  NooBaaBucketClassModel,
+  NooBaaObjectBucketClaimModel,
+} from '../../models';
```
Comment (afreen23) [inline frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx:26]:
```suggestion
import {
  NooBaaBackingStoreModel,
  NooBaaBucketClassModel,
  NooBaaObjectBucketClaimModel,
} from '../../models';
import './activity-card.scss';
```

File: frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx [hunk 6]
```diff
@@ -18,20 +18,37 @@ import {
...
 
+const otherNoobaaEvents = (event: EventKind): boolean => {
+  const eventKind: string = event?.involvedObject?.kind;
+  return (
+    eventKind === NooBaaBackingStoreModel.kind ||
+    eventKind === NooBaaBucketClassModel.kind ||
+    eventKind === NooBaaObjectBucketClaimModel.kind
+  );
+};
+
```
Comment (afreen23) [inline frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx:43]:
```
const noobaaResources = [ NooBaaBackingStoreModel.kind, NooBaaBucketClassModel.kind , NooBaaObjectBucketClaimModel.kind];

File: frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx [hunk 7]
```diff
@@ -18,20 +18,37 @@ import {
...
 const noobaaEventsFilter = (event: EventKind): boolean => {
   const eventKind: string = _.get(event, 'involvedObject.kind');
   if (eventKind === PodModel.kind || eventKind === StatefulSetModel.kind) {
     return isNoobaaEventObject(event);
   }
+  if (otherNoobaaEvents(event)) {
+    return true;
+  }
   return false;
```
Comment (afreen23) [inline frontend/packages/noobaa-storage-plugin/src/components/activity-card/activity-card.tsx:52]:
```suggestion
  return noobaaResources.includes(eventKind)
```

**Comment by a2batic** [comment]
/retitle Bug 1828611: Add events for Noobaa for activity card

**Pull request merged**